src/Plugin.php

```diff
@@ -143,6 +143,9 @@ public static function purgePost($postId) {
    *   The ID of the post the media file is attached to.
    */
   public static function purgeAttachmentMeta($attachment, $postId) {
+    if (get_post_mime_type($postId) === 'application/pdf') {
+      $attachment = wp_get_attachment_metadata($postId);
```

Comment:
1. This callback is registered for the hook `wp_update_attachment_metadata` - why wouldn't $attachment already contain the meta data of the post if it is a PDF?

1. If there is indeed a limitation for attachment (mime) types for which meta data is passed, shouldn't the condition be negated (e.g. not an image) or otherwise more generic?
```suggestion
    if (empty($attachment['file'])) {
      $attachment = wp_get_attachment_metadata($postId);
```

In any case, this could use a code comment as it requires knowledge about internals.